cloudtrail.tf

```diff
@@ -0,0 +1,74 @@
+data "aws_caller_identity" "current" {}
+
+locals {
+  s3_bucket_name = "${var.name}-cloud-trail-bucket"
+}
+resource "aws_cloudtrail" "cloudtrail" {
+  name                          = var.name
```

Comment:
have cloudtrail variables wrapped/grouped into one single object variable instead of top level, so that instead of
var.name
var.create_s3_bucket
var.bucket_name
....
and other variables 

so that you will get them something like this
var.cloudtrail.name
var.cloudtrail.create_s3_bucket
var.cloudtrail.bucket_name
....